core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -489,20 +489,22 @@ SharedAllocationRecord<Kokkos::CudaSpace, void>::attach_texture_object(
 // <editor-fold desc="SharedAllocationRecord destructors"> {{{1
 
 SharedAllocationRecord<Kokkos::CudaSpace, void>::~SharedAllocationRecord() {
-  const char *label = nullptr;
+  std::string label;
   if (Kokkos::Profiling::profileLibraryLoaded()) {
     SharedAllocationHeader header;
     Kokkos::Cuda exec;
-    Kokkos::Impl::DeepCopy<Kokkos::CudaSpace, HostSpace>(
+    Kokkos::Impl::DeepCopy<HostSpace, Kokkos::CudaSpace, Kokkos::Cuda>(
         exec, &header, RecordBase::m_alloc_ptr, sizeof(SharedAllocationHeader));
     exec.fence(
         "SharedAllocationRecord<Kokkos::CudaSpace, "
-        "void>::~SharedAllocationRecord(): fence after copying header from "
+        "void>::~SharedAllocationRecord(): fence after copying header to "
         "HostSpace");
+    // FIXME Why is this deep copy necessary?
```

Comment:
Because otherwise `label.c_str()` that you pass to `deallocate` points to `\0`.

Reply:
Yes, maybe the FIXME should be a little more verbose. It's not clear to me why we need the (deep) copy of the label instead of just copying the pointer.
Before and after this pull request, `label` will only be used if it's modified here.